### PR TITLE
Fix/7938 navigation feedback follow up note should self delete when not relevant

### LINKED ISSUE #7938: NavigationFeedbackFollowUp should self-delete when irrelevant, like NavigationNudge and NavigationFeedback
### Is your feature request related to a problem? Please describe.

In [PR 7914](https://github.com/woocommerce/woocommerce-admin/pull/7914) we added logic to delete the `NavigationNudge` and `NavigationFeedback` notes if the `navigation` feature is not available (which probably means that the woocommerce-admin plugin has been deactivated or is overridden by woo core), but we missed `NavigationFeedbackFollowUp`, which ought to behave the same way.

### Describe the solution you'd like

* Add an implementation of `is_applicable()` to `NavigationFeedbackFollowUp` in `src/Notes/NavigationFeedbackFollowup.php`
* Add `NavigationFeedbackFollowUp::delete_if_not_applicable()` to `Events->possibly_delete_notes()` in `src/Events.php`.

### Should this be prioritized? Why?

On the one hand, this issue is very minor. On the other, it would be very quick to address.

### Additional context

Make sure to review [PR 7914](https://github.com/woocommerce/woocommerce-admin/pull/7914).

### PR DESCRIPTION
**Pull request by jacob-sewell**
Fixes #7938

Applies the same changes we made to `NavigationNudge` and `NavigationFeedback` notes to `NavigationFeedbackFollowUp` note. I.e. it will delete itself as part of the daily wc admin cron if the navigation feature does not exist AND the note has not already been actioned.

### Detailed test instructions:

- Start with a clean WooCommerce site
- Install WooCommerce Admin from this branch and activate it
- Verify that the Navigation checkbox appears on the Settings / Advanced / Features page.
- If the `NavigationNudge` note ('You now have access to the WooCommerce navigation') doesn't show up immediately, run the `wc_admin_daily` cron event. I'm using the WP Crontrol plugin.
- Enable the navigation feature
- If the `NavigationFeedback` note ('You now have access to the new WooCommerce navigation') doesn't show up immediately, run the `wc_admin_daily` cron event.
- Either wait 432,001 seconds or back-date the `NavigationFeedback` note (see below)
- If the `NavigationFeedbackFollowUp` note ('Share your thoughts on the new WooCommerce navigation') doesn't show up immediately, run the `wc_admin_daily` cron event.
- Remove the navigation feature (see below)
- Verify that the `NavigationFeedbackFollowUp` note no longer appears (bonus points: check the database and verify that the only row in `wp_wc_admin_notes` with 'navigation' in its `name` is `wc-admin-navigation-nudge`).
- Reenable the navigation feature: Just undo whatever you did two steps ago.
- Run the `wc_admin_daily` cron event
- Verify that the `NavigationFeedback` note has been recreated
- Back-date the `NavigationFeedback` note
- Run the `wc_admin_daily` cron event
- Verify that the `NavigationFeedbackFollowUp` note has been recreated

#### Backdating the NavigationFeedback note

One way to backdate the `NavigationFeedback` note is to run this query in the database:
`UPDATE wp_wc_admin_notes SET date_created = DATE_ADD(date_created, INTERVAL -6 DAY) WHERE name LIKE 'wc-admin-navigation-%';`

(Yes, this will also backdate the nudge note, but we don't want the poor little fellas getting confused about cause and effect, do we?)

#### One way to remove the navigation feature:

Use the `'woocommerce_admin_features'` filter to remove the `'navigation'` filter like so (you can put this code anywhere it'll be executed--I'm using a custom plugin):

```
add_filter( 'woocommerce_admin_features', function ( $features ) {
	if (!empty($features)) {
		$features = array_filter($features, function ($feature) { return $feature !== 'navigation'; });
	}
	return $features;
} );
```

### CODE REVIEW

File: changelogs/fix-7938-navigation-feedback-follow-up-note-should-self-delete-when-not-relevant [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: Fix
+
+Have NavigationFeedbackFollowUp note self-delete when navigation feature is not present, like the other Navigation notes do. #7939
```
Comment (joshuatf) [inline changelogs/fix-7938-navigation-feedback-follow-up-note-should-self-delete-when-not-relevant:4]: Nitpicky, but I would shorten this to favor brevity in the changelog:

```suggestion
Self-delete NavigationFeedbackFollowUp note when navigation feature is not present. #7939
```

File: src/Notes/NavigationFeedbackFollowUp.php [hunk 2]
```diff
@@ -24,6 +24,13 @@ class NavigationFeedbackFollowUp {
 	 */
 	const NOTE_NAME = 'wc-admin-navigation-feedback-follow-up';
 
+	/**
+	 * Should this note exist? (The navigation feature should exist.)
+	 */
+	public static function is_applicable() {
+		return Features::exists( 'navigation' );
```
Comment (joshuatf) [inline src/Notes/NavigationFeedbackFollowUp.php:31]: If we're going by inverse logic of note creation, we'd want to check if the feature is enabled:

```suggestion
		return Features::is_enabled( 'navigation' );
```

But this note may be useful even if the feature is disabled.  Do we want to still keep this note even if the feature is disabled, @pmcpinto?  Title: "Share your thoughts on the new WooCommerce navigation"

I'm secretly hoping that all notes can just be inverse logic for creation or deletion so we can use `is_applicable` to clean up all note creation/deletion.

**Review by joshuatf — COMMENTED** [review]
Nice follow up, @jacob-sewell!

Left a comment/question; we can wait on @pmcpinto's response to change if necessary.

File: src/Notes/NavigationFeedbackFollowUp.php (hunk 2 above)
Comment (jacob-sewell) [reply src/Notes/NavigationFeedbackFollowUp.php:31]: Oh, this is a good point. My first thought was that this note won't be created unless `NavigationFeedback` is created, which only happens if navigation is enabled, but if they didn't like it and turned it back off we'd want to know that too.

Comment (pmcpinto) [reply src/Notes/NavigationFeedbackFollowUp.php:31]: > But this note may be useful even if the feature is disabled. Do we want to still keep this note even if the feature is disabled, @pmcpinto? Title: "Share your thoughts on the new WooCommerce navigation"

Nice catch. It makes sense to keep the note. Thanks for the suggestion!

Comment (jacob-sewell) [reply src/Notes/NavigationFeedbackFollowUp.php:31]: So where we land is: keep checking `Feature::exists('navigation')` in `is_applicable()`, with apologies to @joshuatf for the asymmetry. Yes?

Comment (joshuatf) [reply src/Notes/NavigationFeedbackFollowUp.php:31]: Yup!  That sounds correct.

**Review by joshuatf — APPROVED** [review]
Looks great and testing well; thanks for the follow-up, Jacob!  :shipit:
